libcaf_core/caf/flow/step/on_error_return.hpp

```diff
@@ -0,0 +1,61 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/detail/type_traits.hpp"
+#include "caf/fwd.hpp"
+
+#include <utility>
+
+namespace caf::flow::step {
+
+template <class T, class Predicate>
```

Comment:
```suggestion
template <class ErrorHandler>
```

A predicate is something that returns a boolean, so I think `ErrorHandler` is a more appropriate name here. Furthermore, we can deduce `T` from the signature of the error handler.